src/System.Security.Cryptography.X509Certificates/tests/X509StoreTests.cs

```diff
@@ -496,5 +499,48 @@ public static void UnixCannotModifyDisallowedStore(bool useEnum, OpenFlags openF
                 Assert.Equal(0, store.Certificates.Count);
             }
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux)] // Windows/OSX doesn't use SSL_CERT_{DIR,FILE}.
+        private void X509Store_MachineStoreLoadSkipsInvalidFiles()
+        {
+            // We create a folder for our machine store and use it by setting SSL_CERT_{DIR,FILE}.
+            // In the store we'll add some invalid files, but we start and finish with a valid file.
+            // This is to account for the order in which the store is populated.
+            string sslCertDir = GetTestFilePath();
+            Directory.CreateDirectory(sslCertDir);
+
+            // Valid file.
+            File.WriteAllBytes(Path.Combine(sslCertDir, "0.pem"), TestData.SelfSigned1PemBytes);
+
+            // File with invalid content.
+            File.WriteAllText(Path.Combine(sslCertDir, "1.pem"), "This is not a valid cert");
+
+            // File which is not readable by the current user.
+            string unreadableFileName = Path.Combine(sslCertDir, "2.pem");
+            File.WriteAllText(unreadableFileName, string.Empty);
+            Assert.Equal(0, chmod(unreadableFileName, 0));
```

Comment:
Is the file going to be deleted as part of cleanup even though it has 000 as perms?

Reply:
Permission of the file doesn't matter. Only the parent dir(s).

While looking at this, I see corefx implementation for Directory.Delete recursive=true is to recursively delete things in the directory. `unlink` of the directory could be an interesting alternative. Do you know if that has been considered?

Reply:
> Do you know if that has been considered?

I don't believe it has.